docs/overview.rst

```diff
@@ -1,380 +1,306 @@
+More complex patterns for creating filters and polling for logs can be found in the
+:ref:`Filtering <filtering>` documentation.
 
-        >>> Web3.soliditySha3(['address'], ["0x49EdDD3769c0712032808D86597B84ac5c2F5614"])
-        HexBytes("0x2ff37b5607484cd4eecf6d13292e22bd6e5401eaffcc07e279583bc742c68882")
 
-        >>> Web3.soliditySha3(['address'], ["ethereumfoundation.eth"])
-        HexBytes("0x913c99ea930c78868f1535d34cd705ab85929b2eaaf70fcd09677ecd6e5d75e9")
+API
+^^^
 
-Check Encodability
-~~~~~~~~~~~~~~~~~~~~
+- :meth:`web3.eth.filter() <web3.eth.Eth.filter>`
+- :meth:`web3.eth.getFilterChanges() <web3.eth.Eth.getFilterChanges>`
+- :meth:`web3.eth.getFilterLogs() <web3.eth.Eth.getFilterLogs>`
+- :meth:`web3.eth.uninstallFilter() <web3.eth.Eth.uninstallFilter>`
+- :meth:`web3.eth.getLogs() <web3.eth.Eth.getLogs>`
+- :meth:`Contract.events.your_event_name.createFilter() <web3.contract.Contract.events.your_event_name.createFilter>`
+- :meth:`Contract.events.your_event_name.build_filter() <web3.contract.Contract.events.your_event_name.build_filter>`
+- :meth:`Filter.get_new_entries() <web3.utils.filters.Filter.get_new_entries>`
+- :meth:`Filter.get_all_entries() <web3.utils.filters.Filter.get_all_entries>`
+- :meth:`Filter.format_entry() <web3.utils.filters.Filter.format_entry>`
+- :meth:`Filter.is_valid_entry() <web3.utils.filters.Filter.is_valid_entry>`
 
-.. py:method:: w3.is_encodable(_type, value)
 
-  Returns ``True`` if a value can be encoded as the given type. Otherwise returns ``False``.
+Net API
+~~~~~~~
 
-   .. code-block:: python
+Some basic network properties are available on the ``web3.net`` object:
 
-        >>> from web3.auto.gethdev import w3
-        >>> w3.is_encodable('bytes2', b'12')
-        True
-        >>> w3.is_encodable('bytes2', b'1')
-        True
-        >>> w3.is_encodable('bytes2', '0x1234')
-        True
-        >>> w3.is_encodable('bytes2', b'123')
-        False
+- :attr:`web3.net.listening`
+- :attr:`web3.net.peer_count`
+- :attr:`web3.net.version`
 
-.. py:method:: w3.enable_strict_bytes_type_checking()
 
-   Enables stricter bytes type checking. For more examples see :ref:`enable-strict-byte-check`
+ethPM
+~~~~~
 
-    .. doctest::
+ethPM allows you to package up your contracts for reuse or use contracts from
+another trusted registry. See the full details :ref:`here <ethpm>`.
 
-        >>> from web3.auto.gethdev import w3
-        >>> w3.enable_strict_bytes_type_checking()
-        >>> w3.is_encodable('bytes2', b'12')
-        True
-        >>> w3.is_encodable('bytes2', b'1')
-        False
 
-Modules
--------
+ENS
+~~~
 
-The JSON-RPC functionality is split across multiple modules which *loosely*
-correspond to the namespaces of the underlying JSON-RPC methods.
+`Ethereum Name Service (ENS) <https://ens.domains/>`_ provides the infrastructure
+for human-readable addresses. As an example, instead of
+``0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045``, you can send funds to ``vitalik.eth``.
```

Comment:
*nit* Super minor: name-checking Vitalik very subtly feeds into a (false) narrative that eth devs worship him, or that he wields some kind of dictatorial power. I would use `ethereumfoundation.eth` or something like it instead.